File: app/src/main/java/com/akay/fitnass/data/db/converter/DateTimeConverter.java
package com.akay.fitnass.data.db.converter;

import android.arch.persistence.room.TypeConverter;

import org.threeten.bp.Instant;
import org.threeten.bp.ZoneId;
import org.threeten.bp.ZonedDateTime;

public class DateTimeConverter {

    @TypeConverter
    public static ZonedDateTime toDate(final long millis) {
        if (millis < 0) {
            return null;
        }
        Instant instant = Instant.ofEpochMilli(millis);
        return ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
    }

    @TypeConverter
    public static long toMillis(final ZonedDateTime zdt) {
        if (zdt == null) {
            return -1L;
        }
        return zdt.toInstant().toEpochMilli();
    }
}

File: app/src/main/java/com/akay/fitnass/data/db/converter/DateTimeConverter.kt
package com.akay.fitnass.data.db.converter

import androidx.room.TypeConverter
import org.threeten.bp.Instant
import org.threeten.bp.ZoneId
import org.threeten.bp.ZonedDateTime

class DateTimeConverter {

    @TypeConverter
    fun toDate(millis: Long): ZonedDateTime = ZonedDateTime.ofInstant(Instant.ofEpochMilli(millis), ZoneId.systemDefault())

    @TypeConverter
    fun toMillis(zdt: ZonedDateTime?): Long = zdt?.toInstant()?.toEpochMilli() ?: 0L
}